paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/PaymentSheetLoader.kt

```diff
@@ -204,14 +204,20 @@ internal class DefaultPaymentSheetLoader @Inject constructor(
             PaymentMethod.Type.fromCode(it.code)
         }
 
-        return customerRepository.getPaymentMethods(
+        val paymentMethods = customerRepository.getPaymentMethods(
             customerConfig = customerConfig,
             types = paymentMethodTypes,
             silentlyFail = true,
-        ).getOrDefault(emptyList()).filter { paymentMethod ->
-            paymentMethod.hasExpectedDetails() &&
-                // PayPal isn't supported yet as a saved payment method (backend limitation).
-                paymentMethod.type != PaymentMethod.Type.PayPal
+        ).getOrDefault(emptyList())
+
+        return paymentMethods.filter { paymentMethod ->
+            paymentMethod.hasExpectedDetails()
+        }.filter { paymentMethod ->
+            // PayPal isn't supported yet as a saved payment method (backend limitation).
+            paymentMethod.type != PaymentMethod.Type.PayPal
```

Comment:
We should (as a follow up) add this as a property to the LpmRepository or similar where we can define this stuff along side the PM.

Reply:
+1